src/pages/buy/buy.layout.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { PopupContainer } from '@components/popup/container';
+import { ChangeScreenAction } from '@common/hooks/use-change-screen';
+import { Header } from '@components/header';
+import { Caption, Text, Title } from '@components/typography';
+import { Button, Flex, Stack } from '@stacks/ui';
+import { Link } from '@components/link';
+import { ScreenPaths } from '@common/types';
+import AddFunds from '@assets/images/add-funds.svg';
+import { SpaceBetween } from '@components/space-between';
+import { openInNewTab } from '@common/utils/open-in-new-tab';
+import { useActiveFiatProviders, useHasFiatProviders } from '@query/hiro-config/hiro-config.query';
+
+const PROVIDERS_INFO = {
+  transak: {
+    title: 'Transak',
+    body: 'International users can purchase STX quickly with EUR, KRW and many more.',
+    cta: 'Buy on Transak',
+  },
+  okcoin: {
+    title: 'Okcoin',
+    body: 'US users can purchase STX quickly with USD',
+    cta: 'Buy on Okcoin',
+  },
+};
+
+interface ProvidersUrl {
+  transak: string;
+  okcoin: string;
+}
+
+interface BuyLayoutProps {
+  doChangeScreen: ChangeScreenAction;
+  providersUrl: ProvidersUrl;
+}
+
+interface ProviderLayoutProps {
+  provider: string;
+  providerUrl: string;
+}
+
+const ProviderLayout = ({ provider, providerUrl }: ProviderLayoutProps) => {
+  const { title, cta, body } = PROVIDERS_INFO[provider as keyof ProvidersUrl];
+  return (
+    <Stack
+      overflow="hidden"
+      alignItems="flex-start"
+      spacing="base-tight"
+      padding="24px"
+      mt={5}
+      className="buy-box"
+    >
+      <SpaceBetween flexGrow={1}>
+        <Stack spacing="base-tight">
+          <Title marginBottom="10">{title}</Title>
+          <Caption>{body}</Caption>
+        </Stack>
+      </SpaceBetween>
+      <Button width="100%" mt={5} onClick={() => openInNewTab(providerUrl)} borderRadius="10px">
+        {cta}
+      </Button>
+    </Stack>
+  );
+};
+
+export const BuyLayout = (props: BuyLayoutProps) => {
+  const { doChangeScreen, providersUrl } = props;
+  const activeProviders = useActiveFiatProviders();
+  const hasProviders = useHasFiatProviders();
+  if (!hasProviders) return null;
+
+  return (
+    <PopupContainer
+      header={
+        <Header hideActions title=" " onClose={() => doChangeScreen(ScreenPaths.POPUP_HOME)} />
```

Comment:
Rather than passing in the function by reference, why not create an event prop: `onCloseAction()`

```tsx
<BuyLayout onCloseAction={() => changeScreen(ScreenPaths.POPUP_HOME)} />
```